pkg/ccl/backupccl/restore_online.go

```diff
@@ -112,6 +118,29 @@ func sendAddRemoteSSTWorker(
 		var batchSize int64
 		const targetBatchSize = 440 << 20
 
+		// TODO(dt): handle tenants
+		prefixes := prefixRewriter{rewrites: make([]prefixRewrite, 0, len(tableRekeys))}
+		for _, i := range tableRekeys {
+
+			if i.OldID == 0 {
+				continue
+			}
```

Comment:
This is copied from restore data processor; currently restore processors / users of rewrite rules are expected to skip rules with oldID=0, so we're doing so here.

Backstory: at one point we needed to backport a fix which changed semantics of rewrites -- how we generate them and how we use them (IIRC we started adding rewrites that were not there before but needed to be handled specially or else it would cause incorrectness).

To backport, this change couldn't use a version gate, but we still needed to be sure that only updated processors/users of rewrites acted on the modified rewrites, so we started including a special rewrite rule with ID 0 (and a descriptive name in its value) so only users of rewrites who had been updated to skip id=0 would succeed, while others would error out.

I think with minVersion bumped we could now remove this -- from all restores -- but that should be its own change. For now we're just sticking with expected usage of rewrites.

Reply:
Ah gotcha, this is the happy path on the upgraded side for that poison pill path.